### PR TITLE
Added a lot of Functionality and Config Options

### PR DESCRIPTION
**Pull request by NickReagan**
- Added IsDisplaysHidden Export
- Added AOP Change Announcement with {NEW_AOP} placeholder.
- Added Respawn Locations that work off of the set AOP and sends you to a default location if the set aop doesn't match one in the table. 
- Added AcePerm options for each command in config.
- Added some more logic code for commands.
- Moved postals to a separate file to avoid corruption when just making small edits to config with most FTPs.

### CODE REVIEW

**Comment by NickReagan** [comment]
I understand if it isn't pulled since this adds a lot.
Just made some edits for my server and figured I would attempt to contribute them. <3

File: client.lua [hunk 1]
```diff
@@ -144,15 +149,33 @@ AddEventHandler('Badssentials:RevivePlayer', function()
 	local ped = GetPlayerPed(-1);
 	if IsEntityDead(ped) then 
 		revivePed(ped);
+		TriggerEvent('chat:addMessage', {args = {Config.Prefix .. "Revived successfully!"} });
```
Comment (JaredScar) [inline client.lua]: Can we make such messages configurable? Probably better to have this configurable than to be plain like this.

File: client.lua [hunk 2]
```diff
@@ -144,15 +149,33 @@ AddEventHandler('Badssentials:RevivePlayer', function()
...
+		local foundRespawnLocation = nil
+
+		--loops through respawn locations and finds one that matches current AOP
+		for i, v in pairs(Config.ReviveSystem.RespawnLocations) do
+			if i == currentAOP then
+				foundRespawnLocation = true
+				revivePed(ped);
+				SetEntityCoords(ped, v.x, v.y, v.z, false, false, false, false);
+				SetEntityCoords(entity, xPos, yPos, zPos, xAxis, yAxis, zAxis, clearArea)
+				TriggerEvent('chat:addMessage', {args = {Config.Prefix .. "Respawned successfully!"} });
```
Comment (JaredScar) [inline client.lua]: Configurable message?

File: client.lua [hunk 3]
```diff
@@ -144,15 +149,33 @@ AddEventHandler('Badssentials:RevivePlayer', function()
...
+			end
+		end
+
+		--sends player to defualt spawn location if currentAOP doesn't match when in the table.
+		if foundRespawnLocation ~= true then
+			revivePed(ped);
+			SetEntityCoords(ped, Config.ReviveSystem.RespawnLocations.DefaultLocation.x, Config.ReviveSystem.RespawnLocations.DefaultLocation.y, Config.ReviveSystem.RespawnLocations.DefaultLocation.z, false, false, false, false);
+			SetEntityCoords(entity, xPos, yPos, zPos, xAxis, yAxis, zAxis, clearArea)
+			TriggerEvent('chat:addMessage', {args = {Config.Prefix .. "Respawned successfully!"} });
```
Comment (JaredScar) [inline client.lua]: Configurable message?

File: server.lua [hunk 4]
```diff
@@ -1,17 +1,53 @@
...
 end
 
+function GetPeaceTimeStatus()
+  return peacetime
+end
+
 RegisterCommand(Config.ScreenAffects.AnnounceCommand, function(source, args, raw) 
   local src = source;
-  if IsPlayerAceAllowed(src, "Badssentials.Announce") then 
+  if IsPlayerAceAllowed(src, Config.ScreenAffects.acePermission) then
```
Comment (JaredScar) [inline server.lua]: `AcePermission` would probably be a better naming scheme

File: server.lua [hunk 5]
```diff
@@ -49,16 +85,25 @@ peacetime = false;
 currentAOP = Config.AOPSystem.DefaultAOP; -- By default 
 RegisterCommand(Config.AOPSystem.AOPCommand, function(source, args, rawCommand)
   local src = source;
-  if IsPlayerAceAllowed(src, "Badssentials.AOP") then 
+  if IsPlayerAceAllowed(src, Config.AOPSystem.acePermission) then
```
Comment (JaredScar) [inline server.lua:93]: `AcePermission` would probably be a better naming scheme

File: server.lua [hunk 6]
```diff
@@ -49,16 +85,25 @@ peacetime = false;
...
-  if IsPlayerAceAllowed(src, "Badssentials.AOP") then 
+  if IsPlayerAceAllowed(src, Config.AOPSystem.acePermission) then 
     -- Allowed to use /aop <aop>
     if #args > 0 then 
       currentAOP = table.concat(args, " ");
       sendMsg(src, "You have set the AOP to: " .. currentAOP);
       TriggerClientEvent('Badssentials:SetAOP', -1, currentAOP);
+
+      if Config.AOPSystem.aopAnnouncement ~= nil then
+        local aopAnnouncement = Config.AOPSystem.aopAnnouncement
```
Comment (JaredScar) [inline server.lua]: `AOP_Announcement` is probably preferred

File: server.lua [hunk 7]
```diff
@@ -88,29 +133,41 @@ AddEventHandler("Badssentials:DeathTrigger", function()
 end)
 RegisterCommand(Config.ReviveSystem.ReviveCommand, function(source, args, rawCommand)
   local src = source;
-  if #args == 0 then 
+  if #args == 0 or tonumber(args[1]) == src then 
     -- Revive themselves
     if timersRev[src] ~= nil and timersRev[src] >= 0 then 
       -- They are dead and have a timer 
-      if IsPlayerAceAllowed(src, "Badssentials.Bypass.Revive") then 
+      if IsPlayerAceAllowed(src, Config.ReviveSystem.bypassReviveAcePermission) then
```
Comment (JaredScar) [inline server.lua]: `BypassReviveAcePermission`

File: server.lua [hunk 8]
```diff
@@ -88,29 +133,41 @@ AddEventHandler("Badssentials:DeathTrigger", function()
...
-      TriggerClientEvent('Badssentials:RevivePlayer', tonumber(args[1]));
-      sendMsg(src, "You have revived player ^5" .. GetPlayerName(tonumber(args[1])) .. " ^3successfully!");
-      sendMsg(tonumber(args[1]), "You have been revived successfully by ^5" .. GetPlayerName(src) .. "^3!");
+    if IsPlayerAceAllowed(src, Config.ReviveSystem.reviveOthersAcePermission) then
+      --Checks if arg is number
+      if IsInt(args[1]) then 
+        --checks if a player with ID is online
+        if IsPlayerOnline(args[1]) then
+          TriggerClientEvent('Badssentials:RevivePlayer', tonumber(args[1]));
+          sendMsg(src, "You have revived player ^5" .. GetPlayerName(tonumber(args[1])) .. " ^3successfully!");
```
Comment (JaredScar) [inline server.lua]: Configurable message?

File: server.lua [hunk 9]
```diff
@@ -88,29 +133,41 @@ AddEventHandler("Badssentials:DeathTrigger", function()
...
+          sendMsg(tonumber(args[1]), "You have been revived successfully by ^5" .. GetPlayerName(src) .. "^3!");
```
Comment (JaredScar) [inline server.lua]: Configurable message?

File: server.lua [hunk 10]
```diff
@@ -120,26 +177,24 @@ RegisterCommand(Config.ReviveSystem.RespawnCommand, function(source, args, rawCo
...
-      sendMsg(src, "You have respawned successfully!");
     end
   end 
 end)
 
 
 RegisterCommand(Config.Misc.Peacetime, function(source, args, rawCommand)
   local src = source;
-  if IsPlayerAceAllowed(src, "Badssentials.PeaceTime") then
+  if IsPlayerAceAllowed(src, Config.Misc.ptAcePermission) then
```
Comment (JaredScar) [inline server.lua]: `PeacetimeAcePermission`

**Comment by NickReagan** [comment]
I will get those done and do another commit.

File: client.lua [hunk 11]
```diff
@@ -112,72 +112,76 @@ AddEventHandler('onClientMapStart', function()
...
+
+if Config.ReviveSystem.enable then
+	deadCheck = false;
+	Citizen.CreateThread(function()
+		while true do 
+			Wait(0);
+			local ped = GetPlayerPed(-1);
+			if Config.ScreenAffects.DeathScreen then
+				if IsEntityDead(ped) then 
+					Draw2DText(.5, .3, "~r~You are knocked out or dead...", 1.0, 1);
```
Comment (JaredScar) [inline client.lua]: It would be nice if these were configurable too

File: server.lua [hunk 12]
```diff
@@ -181,12 +195,22 @@ RegisterCommand(Config.ReviveSystem.RespawnCommand, function(source, args, rawCo
...
-        sendMsg(src, '^1ERROR: You cannot respawn, you still have ^7' .. timersRes[src] .. ' ^1seconds remaining...');
+        local timeLeft = timersRes[src]
+        if timeLeft >= 60 then
+          timeLeft = math.ceil(timeLeft / 60) .. " minute(s)" 
+        else
+          timeLeft = timeLeft .. " seconds"
+        end
+
+        local errorMessage = Config.ReviveSystem.RespawnErrorMessage
+        errorMessage = errorMessage:gsub("{TIME_LEFT}", timeLeft)
```
Comment (JaredScar) [inline server.lua]: `{RESPAWN_TIME_LEFT}` is probably a better naming idea

**Comment by NickReagan** [comment]
[a70602b](https://github.com/JaredScar/Badssentials/pull/8/commits/a70602b7bfc72b48e6b97d11f00eb4a7bd0d1210) Added settings for the death screen as well as some useful placeholders.

**Comment by JaredScar** [comment]
Thanks for all the changes!

**Pull request merged**